src/modules/routing/services/MediationRecipientService.ts

```diff
@@ -0,0 +1,264 @@
+  KeylistUpdateAction,
+  ForwardMessage,
+  MediationGrantMessage,
+  MediationDenyMessage,
+  RequestMediationMessage,
+  KeylistUpdateResponseMessage,
+} from '../messages'
+import { Logger } from '../../../logger'
+import { EventEmitter } from 'events'
+import { Repository } from '../../../storage/Repository'
+import { ConnectionInvitationMessage, ConnectionRecord } from '../../connections'
+import { MediationEventType, MediationKeylistEvent, MediationStateChangedEvent } from './MediationService'
+import { InboundMessageContext } from '../../../agent/models/InboundMessageContext'
+import { OutboundMessage } from '../../../types'
+import { isIndyError } from '../../../utils/indyError'
+import { DefaultMediationRecord, MediationRecord, MediationRecordProps, MediationRole, MediationState, MediationStorageProps } from '..'
+import { Wallet } from '../../../wallet/Wallet'
+
+export enum MediationRecipientEventType {
+  Granted = 'GRANTED',
+  Denied = 'DENIED',
+  KeylistUpdated = 'KEYLIST_UPDATED',
+}
+
+export class MediationRecipientService extends EventEmitter {
+  private agentConfig: AgentConfig
+  private mediatorRepository: Repository<MediationRecord>
+  private messageSender: MessageSender
+  private defaultMediator?: DefaultMediationRecord
+  private wallet: Wallet
+
+  public constructor(
+    agentConfig: AgentConfig,
+    mediatorRepository: Repository<MediationRecord>,
+    messageSender: MessageSender,
+    wallet: Wallet
+  ) {
+    super()
+    this.agentConfig = agentConfig
+    this.mediatorRepository = mediatorRepository
+    this.messageSender = messageSender
+    this.wallet = wallet
+    this.provision()
+  }
+
+  private provision() {
+    /* TODO: handle config flag behaviors.
+    openMediation
+                "Enables didcomm mediation. After establishing a connection, "
+                "if enabled, an agent may request message mediation, which will "
+                "allow the mediator to forward messages on behalf of the recipient. "
+                "See aries-rfc:0211."
+    mediatorInvitation
+                "Connect to mediator through provided invitation "
+                "and send mediation request and set as default mediator."
+    //mediatorConnectionsInvite
+                //--"Connect to mediator through a connection invitation. "
+                //"If not specified, connect using an OOB invitation. "
+                //"Default: false."--
+    defaultMediatorId
+                "Set the default mediator by ID"
+    clearDefaultMediator
+                "Clear the stored default mediator."
+    */
+    // check for default mediation id record
+      // set this.defaultMediator
+    // else if mediation invitation in config
+      // Use agent config to establish connection with mediator
+      // request mediation record
+      // Upon granting, set this.defaultMediator
+  }
+
+  public async create({ state, role, connectionId, recipientKeys }: MediationRecordProps): Promise<MediationRecord> {
```

Comment:
`this.mediationRecipentService.createMediationRecord` feels to long, how about `createRecord`?
I will change service naming to follow module naming, then it will look something like. 
`this.recipentService.createRecord` or inside the service, `this.createRecord`. Compromised?